packages/maker/src/api/defineInterfaceLinkConstraint.ts

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2024 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import invariant from "tiny-invariant";
+import type { InterfaceType } from "./types.js";
+
+type Meta = { apiName: string; displayName?: string; description?: string };
+type ApiNameOrMeta = string | Meta;
+
+type Many = { type: InterfaceType; many: ApiNameOrMeta; one?: never };
+type One = { type: InterfaceType; one: ApiNameOrMeta; many?: never };
+
+export function defineInterfaceLinkConstraint(
```

Comment:
I'm confused, do you still need these functions? (lines 26-36)

Reply:
Yes those are overloads. We don't support many-to-many. I could rewrite this I guess.

Reply:
Ok but the third function you wrote with the invariants throw an error for many-to-many right? Did you just want to overload it so that the function params also reflect that restriction?

Reply:
Yeah I wanted to be sure if you called it from JS or did an `any` that it was helpful. And I wanted the types to reflect it. But I rewrote it in my update (about to push) since it can just be a union.